packages/amplify_flutter/lib/src/config/auth/cognito/mfa.dart

```diff
@@ -0,0 +1,35 @@
+//
+// Copyright 2021 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License").
+// You may not use this file except in compliance with the License.
+// A copy of the License is located at
+//
+//  http://aws.amazon.com/apache2.0
+//
+// or in the "license" file accompanying this file. This file is distributed
+// on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+// express or implied. See the License for the specific language governing
+// permissions and limitations under the License.
+//
+
+import 'package:json_annotation/json_annotation.dart';
+
+enum MfaType {
+  @JsonValue('SMS')
+  sms,
+
+  @JsonValue('TOTP')
+  totp,
+}
+
+enum MfaConfiguration {
+  @JsonValue('OPTIONAL')
+  optional,
+
+  @JsonValue('ON')
+  on,
```

Comment:
Any issues with using a reserved word here?

Reply:
Nope, not sure why not - maybe because `on` is not a valid keyword within a class/enum scope, only in the mixin declartion scope.

Reply:
Alright, as long as we've tested.

Reply:
Yeah, some keywords do not cause issues, but I think it depends on the scope still. Like you can use `factory` as a variable name no problem: https://github.com/dnys1/amplify-flutter/blob/authenticator/common-config/packages/amplify_flutter/lib/src/config/amplify_plugin_registry.dart#L91